test/framework/src/main/java/org/elasticsearch/common/util/concurrent/DeterministicTaskQueue.java

```diff
@@ -52,7 +52,7 @@ public class DeterministicTaskQueue {
     private final List<Runnable> runnableTasks = new ArrayList<>();
     private final Random random;
     private List<DeferredTask> deferredTasks = new ArrayList<>();
-    private long currentTimeMillis;
+    private long currentTimeMillis = System.currentTimeMillis();
```

Comment:
I don't follow why we need this change here. Really `currentTimeMillis` is a _relative_ time, so its absolute value shouldn't matter. I'm +1 on making it not start from zero, in fact I've wanted to do that for a while, but I'd rather it started from a random value (derived from the test seed) rather than the absolute current time since this is not repeatable.

Also I suggest breaking this change out into a separate PR. I expect that it _should_ be independent of this one, at least if it isn't then that suggests we have a bug somewhere.

Reply:
Restored original logic for DeterministicTaskQueue to use 0 as a time basis for task execution. In initial version of PR the time basis was changed to deal with timing issue in SnapshotResiliencyTests#testSnapshotPrimaryRelocations. Root cause is described in #95800. Whilst #95800 is not implemented to fix the test, custom timing adjustment is done, see changes to SnapshotResiliencyTests around L1619